datafusion/optimizer/Cargo.toml

```diff
@@ -53,5 +53,6 @@ log = { workspace = true }
 regex-syntax = "0.8.0"
 [dev-dependencies]
 ctor = { workspace = true }
+datafusion-functions = { workspace = true }
```

Comment:
Let's not do this -- we really need to avoid re-introducing dependencies like this

Tests that need functions should go in https://github.com/apache/datafusion/blob/main/datafusion/core/tests/optimizer_integration.rs I think

Reply:
> Let's not do this -- we really need to avoid re-introducing dependencies like this
> 
> Tests that need functions should go in https://github.com/apache/datafusion/blob/main/datafusion/core/tests/optimizer_integration.rs I think

I'll need to make `simplify_inner` public in order to move the tests. Otherwise I need to somehow pull the functions from some existing dependency but I didn't find anything that re-exports them.

Reply:
Alternatively I just don't check the iteration count on the test case that uses UDFs and move it to integration tests alone.

Reply:
> I'll need to make simplify_inner public in order to move the tests

I think that would be fine too as long as the doc comments explain that